### PR TITLE
ITL-10 Create CRUD operations

### PR DESCRIPTION
**Pull request by Fekete-Balazs710**

### CODE REVIEW

File: client/src/router/router.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import Signup from '../views/Signup.vue'
+
+//Creating routes for multiple pages
+const router = createRouter({
+    history: createWebHistory(),
+    routes:[
+        { path: '/login', component: Login},
+        { path: '/signup', component: Signup},
+        { path: '/dashboard', component: Dashboard},
+        { path: "/", component: Login}
```
Comment (SzaszWebgurus) [inline client/src/router/router.ts]: use redirect here

File: client/src/service/todo.Delete.ts
Comment (SzaszWebgurus) [inline client/src/service/todo.Delete.ts:1]: merge all service micro files into a single todo-service.ts file, which exports a TodoService instance, with the service functions

**Review by SzaszWebgurus — CHANGES_REQUESTED** [review]

File: server/service/todoService.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                { new: true }
+            );
+            res.json(tUpdate);
+        } catch (error) {
+            console.error(error);
+            res.status(500).json({ message: "Failed to update todo's description" });
+        }
+    }
+
+    async updateIsEditing(req, res) {
```
Comment (SzaszWebgurus) [inline server/service/todoService.js]: this is only a temporary state, and shouldn't be saved in the database

File: server/service/todoService.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import TodoModel from "../model/todosModel.js";
+
+class TodoService {
+    async deleteTodo(req, res) {
```
Comment (SzaszWebgurus) [inline server/service/todoService.js]: just delete, don't need to name it like deleteTodo, because the context is given by the class name

Comment (SzaszWebgurus) [inline server/service/todoService.js]: the request and response object shouldn't be here, only in the controllers...here you need only the todoId as parameter...apply this logic to each service function

File: server/service/todoService.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                date: Date.now()
+            });
+            const savedTodo = await newTodo.save();
+            res.json(savedTodo);
+        } catch (error) {
+            res.status(500).json({ error: "Error creating new todo" });
+        }
+    }
+
+    async updateIsChecked(req, res) {
```
Comment (SzaszWebgurus) [inline server/service/todoService.js]: a single generic update method is more than enough for now

File: server/model/todosModel.js [hunk 5]
```diff
@@ -24,10 +24,18 @@ const todoSchema = new mongoose.Schema({
...
-        required: true
+    isEditing: {
+        type: Boolean,
+        default: true
+    },
+    date: {
+        type: Date,
+        default: Date.now()
     }
+    // userId: {
```
Comment (SzaszWebgurus) [inline server/model/todosModel.js]: remove unused code

**Review by SzaszWebgurus — CHANGES_REQUESTED** [review]

**Comment by tamasm-wg** [comment]
Is there any reason you are using .js files? Every file (models, controllers, services, etc. ) should be typescript files.

**Pull request closed without merging**